jsonschema/util.go

```diff
@@ -455,6 +455,46 @@ func fieldJSONInfo(f reflect.StructField) jsonInfo {
 	return info
 }
 
+// walkStructProperties traverses a struct's fields using the same precedence rules as encoding/json.
+// It calls the provided visit function for each discovered field. The StructField passed to the
+// visit function will have its Index field populated with the full path from the top-level struct.
+func walkStructProperties(t reflect.Type, visit func(sf reflect.StructField) error) error {
```

Comment:
Nit: it feels like this helper really belongs in infer.go, since it's directly related to inference. I don't see it being used elsewhere, so it's not really a utility.

Reply:
Or nevermind: I don't think this function is necessary: you can just use reflect.VisibleFields.

Reply:
Good point. Done